src/modules/cpu/kernel/glitch.hpp

```diff
@@ -0,0 +1,1133 @@
+    {
+        __m128i maskR = _mm_setr_epi8(0, 0x80, 0x80, 3, 0x80, 0x80, 6, 0x80, 0x80, 9, 0x80, 0x80, 12, 0x80, 0x80, 15);
+        __m128i maskGB = _mm_setr_epi8(0x80, 1, 2, 0x80, 4, 5, 0x80, 7, 8, 0x80, 10, 11, 0x80, 13, 14, 0x80);
+        RpptROI roi;
+        RpptROIPtr roiPtrInput = &roiTensorPtrSrc[batchCount];
+        compute_roi_validation_host(roiPtrInput, &roi, &roiDefault, roiType);
+
+        Rpp32u xOffsetRchn = xOffsetR[batchCount];
+        Rpp32u yOffsetRchn = yOffsetR[batchCount];
+        Rpp32u xOffsetGchn = xOffsetG[batchCount];
+        Rpp32u yOffsetGchn = yOffsetG[batchCount];
+        Rpp32u xOffsetBchn = xOffsetB[batchCount];
+        Rpp32u yOffsetBchn = yOffsetB[batchCount];
+
+        Rpp32u elementsInRowMax = srcDescPtr->w;
+
+        Rpp32u xOffsets[3] = {xOffsetRchn, xOffsetGchn, xOffsetBchn};
+        Rpp32u yOffsets[3] = {yOffsetRchn, yOffsetGchn, yOffsetBchn};
+        Rpp32u xOffsetsLoc[3] = {xOffsetRchn, xOffsetGchn, xOffsetBchn};
+        Rpp32u yOffsetsLoc[3] = {yOffsetRchn * elementsInRowMax, yOffsetGchn * elementsInRowMax, yOffsetBchn * elementsInRowMax};
+
+        Rpp8u *srcPtrImage, *dstPtrImage;
+        srcPtrImage = srcPtr + batchCount * srcDescPtr->strides.nStride;
+        dstPtrImage = dstPtr + batchCount * dstDescPtr->strides.nStride;
+
+        Rpp32u bufferLength = roi.xywhROI.roiWidth * layoutParams.bufferMultiplier;
+        Rpp8u *srcPtrChannel, *dstPtrChannel;
+        srcPtrChannel = srcPtrImage + (roi.xywhROI.xy.y * srcDescPtr->strides.hStride) + (roi.xywhROI.xy.x * layoutParams.bufferMultiplier);
+        dstPtrChannel = dstPtrImage;
+
+        if ((srcDescPtr->c == 3) && (srcDescPtr->layout == RpptLayout::NHWC) && (dstDescPtr->layout == RpptLayout::NCHW ))
+        {
+            convert_Pkd3_Pln3<Rpp8u>(srcPtrImage, dstPtrImage, srcDescPtr, dstDescPtr, roi.xywhROI.roiWidth * 3);
+            Rpp32u vectorIncrement = 32;
+            for(int c = 0; c < 3; c++)
+            {
+                Rpp32u offset = yOffsetsLoc[c] + xOffsetsLoc[c];
+                Rpp8u *srcPtrImageTemp, *dstPtrImageTemp;
+                srcPtrImageTemp = dstPtrChannel + (c * dstDescPtr->strides.cStride) + offset;
+                dstPtrImageTemp = dstPtrImage + (c * dstDescPtr->strides.cStride);
+                if(offset > 0)
+                {
+                    for(int j = yOffsets[c]; j < roi.xywhROI.roiHeight; j++)
+                    {
+                        Rpp8u *srcTemp = srcPtrImageTemp;
+                        Rpp8u *dstTemp = dstPtrImageTemp;
+                        Rpp32u aligLen = ((roi.xywhROI.roiWidth - xOffsetsLoc[c]) / vectorIncrement) * vectorIncrement;
+                        Rpp32u remElemLen = roi.xywhROI.roiWidth - ( aligLen + xOffsetsLoc[c]);
+                        if(offset < dstDescPtr->strides.cStride)
+                        {
+                            for(int i = 0; i < aligLen; i += vectorIncrement)
```

Comment:
Please rename 
aligLen to alignedLength
remElemLen to remainingElementLength / remainingLength